src/main/java/org/fenixedu/academicextensions/services/registrationhistory/RegistrationHistoryReport.java

```diff
@@ -989,12 +1009,56 @@ public BigDecimal getTuitionAmount() {
 
         final IAcademicTreasuryEvent event =
                 treasuryBridgeAPI.getTuitionForRegistrationTreasuryEvent(registration, getExecutionYear());
+        if (event == null) {
+            return BigDecimal.ZERO;
+        }
+
+        return event.getAmountWithVatToPay().add(event.getNetExemptedAmount());
+    }
+
+    public BigDecimal getPaidTuitionAmount() {
+        final ITreasuryBridgeAPI treasuryBridgeAPI = TreasuryBridgeAPIFactory.implementation();
+        if (treasuryBridgeAPI == null) {
+            return BigDecimal.ZERO;
+        }
+
+        final IAcademicTreasuryEvent event =
+                treasuryBridgeAPI.getTuitionForRegistrationTreasuryEvent(registration, getExecutionYear());
```

Comment:
Since getting the event is always the same in all these 4 methods, we could store it as a field (and only load it when it's not null)